qcodes/instrument_drivers/AlazarTech/ATS.py

```diff
@@ -445,6 +445,36 @@ def acquire(self, mode=None, samples_per_record=None,
         samples_per_record = self.samples_per_record.raw_value
         records_per_buffer = self.records_per_buffer.raw_value
 
+        # bits per sample
+        _, bits_per_sample = self.api.get_channel_info_(self._handle)
+        # Whole bytes per sample
+        bytes_per_sample = (bits_per_sample + 7) // 8
+        # bytes per record
+        bytes_per_record = bytes_per_sample * samples_per_record
+
+        # channels
+        channels_binrep = self.channel_selection.raw_value
+        number_of_channels = self.get_num_channels(channels_binrep)
+
+        # bytes per buffer
+        bytes_per_buffer = (bytes_per_record *
+                            records_per_buffer * number_of_channels)
+
+        sample_type = ctypes.c_uint16 if bytes_per_sample > 1 else ctypes.c_uint8
+
+        max_buffer_size = 64*1024*1024
+        # 64 MB see docs of AlazarBeforeAsyncRead
+        # http://www.alazartech.com/Support/Download%20Files/ATS-SDK-Guide-7.2.3.pdf#section*.110
+
+        requested_buffer_size = (bits_per_sample * samples_per_record *
```

Comment:
and an extra question that is related to the above, what are the meanings of `requested_buffer_size` and `bytes_per_buffer` variables? (The names are not helping me (( ). My initial expectation is that `requested_buffer_size == bytes_per_buffer` and hence is not needed. What am i missing?

Reply:
when i read a C++ example of DualPort NPT acquisition for ATS9360, i see the formula that `bytesPerBuffer = bytesPerRecord * recordsPerBuffer * channelCount` - channel number is involved :)

Reply:
A variation of the above

Reply:
still don't get why "internal buffer" (the on-board-memory buffer) size limit does not take number of channels into account :(